ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -60,10 +61,43 @@ type RayActorOptionSpec struct {
 	AcceleratorType   string                 `json:"accelerator_type,omitempty"`
 }
 
-// ServeDeploymentStatuses defines the current states of all Serve Deployments.
+type AppStatusV1 struct {
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+type ServeDeploymentStatusV1 struct {
+	Name    string `json:"name,omitempty"`
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+// (V1) ServeDeploymentStatuses defines the current states of all Serve Deployments.
 type ServeDeploymentStatuses struct {
-	ApplicationStatus  rayv1alpha1.AppStatus               `json:"app_status,omitempty"`
-	DeploymentStatuses []rayv1alpha1.ServeDeploymentStatus `json:"deployment_statuses,omitempty"`
+	ApplicationStatus  AppStatusV1               `json:"app_status,omitempty"`
+	DeploymentStatuses []ServeDeploymentStatusV1 `json:"deployment_statuses,omitempty"`
+}
+
+type ServeDeploymentDetails struct {
+	Name    string `json:"name"`
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type ServeApplicationDetails struct {
+	Name                    string                            `json:"name"`
+	RoutePrefix             string                            `json:"route_prefix"`
+	DocsPath                string                            `json:"docs_path"`
+	Status                  string                            `json:"status"`
+	Message                 string                            `json:"message"`
+	LastDeployedTimeSeconds float32                           `json:"last_deployed_time_s"`
```

Comment:
I am not sure how `LastDeployedTimeSeconds` is generated and what its precision is in Ray's Python code. However, the following code snippet shows that it takes 24 bytes on my devbox. Is it sufficient to use float32, or is there a possibility of overflow?

```python
import sys
import time

# Create a time object
t = time.time()

# Get the size of the time object in bytes
size = sys.getsizeof(t)

print(f"Size of time object: {size} bytes")
```

Reply:
Just realized we don't currently need this information, so I've removed it from the struct. Not sure about your question specifically, if we add it back in perhaps we should use float64 instead.

Reply:
After conducting some research, I have come to the conclusion that in most cases, a `float` in Python is 64 bits. Hence, `float64` may be better than `float32`.